pkg/pipeline/resource_pipeline_node_pool.go

```diff
@@ -0,0 +1,212 @@
+			Required:    true,
+			Description: "Set to true for dynamic node pool. Set to false for static node pool.",
+		},
+		"architecture": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "Set the architecture. This is currently limited to x86_64.",
+		},
+		"operating_system": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "Operating systems supported for the selected architecture.",
+		},
+		"node_idle_interval_in_mins": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+			Description:  "Number of minutes a node can be idle before it is destroyed.",
+		},
+		"environments": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			Description: "In a project, an array of environment names in which this pipeline source will be.",
+		},
+	}
+
+	var unpackNodePool = func(data *schema.ResourceData) (NodePool, error) {
+		d := &ResourceData{data}
+
+		nodePool := NodePool{
+			ProjectId:              d.getInt("project_id"),
+			Name:                   d.getString("name"),
+			NumberOfNodes:          d.getInt("number_of_nodes"),
+			IsOnDemand:             d.getBool("is_on_demand"),
+			Architecture:           d.getString("architecture"),
+			OperatingSystem:        d.getString("operating_system"),
+			NodeIdleIntervalInMins: d.getInt("node_idle_interval_in_mins"),
+			Environments:           d.getList("environments"),
+		}
+		return nodePool, nil
+	}
+
+	var packNodePool = func(d *schema.ResourceData, nodePool NodePool) diag.Diagnostics {
+		var errors []error
+		setValue := mkLens(d)
+
+		errors = setValue("project_id", nodePool.ProjectId)
+		errors = append(errors, setValue("name", nodePool.Name)...)
+		errors = append(errors, setValue("number_of_nodes", nodePool.NumberOfNodes)...)
+		errors = append(errors, setValue("is_on_demand", nodePool.IsOnDemand)...)
+		errors = append(errors, setValue("architecture", nodePool.Architecture)...)
+		errors = append(errors, setValue("operating_system", nodePool.OperatingSystem)...)
+		errors = append(errors, setValue("node_idle_interval_in_mins", nodePool.NodeIdleIntervalInMins)...)
+		errors = append(errors, setValue("environments", nodePool.Environments)...)
+
+		if len(errors) > 0 {
+			return diag.Errorf("failed to pack node pool %q", errors)
+		}
+
+		return nil
+	}
+
+	var readNodePool = func(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+		log.Printf("[DEBUG] readNodePool")
+
+		// This one returns an array of nodepools because the api doesn't seem to provide a GET
+		// for a single node pool id. instead it's a query value on id.
+
+		nodePools := []NodePool{}
+		resp, err := m.(*resty.Client).R().
+			SetResult(&nodePools).
+			Get(nodePoolsUrl + "?nodePoolIds=" + data.Id())
```

Comment:
I'd encourage using Resty's `SetQueryParams()` https://pkg.go.dev/github.com/go-resty/resty/v2#Request.SetQueryParam for `nodePoolIds`